### PR TITLE
Basic auth for https

### PR DESCRIPTION
**Pull request by iffyuva**
#### What this PR does / why we need it:

This PR adds basic authentication to entrypoints marked as https. Sometimes we need to protect our internal services via basic auth.

#### Which issue this PR fixes
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
    - fixes #

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] [DCO](https://www.helm.sh/blog/helm-dco/index.html) signed
- [x] Chart Version bumped
- [x] Variables are documented in the README.md

### CODE REVIEW

**Comment by cpanato** [comment]
@iffyuva please bump the chart version and also sign the DCO

File: stable/traefik/Chart.yaml [hunk 1]
```diff
@@ -1,6 +1,6 @@
 apiVersion: v1
 name: traefik
-version: 1.43.0
+version: 1.43.1
```
Comment (cpanato) [inline stable/traefik/Chart.yaml]: new functionality should be 1.44.0

**Comment by iffyuva** [comment]
Can we merge this in? /cc @Juliens @nmengin @cpanato

**Pull request merged**